general/src/store/normalize/user_agent.rs

```diff
@@ -47,13 +47,30 @@ pub fn normalize_user_agent(event: &mut Event) {
         })));
     }
 
-    if is_known(os.family.as_str()) && !contexts.contains_key(OsContext::default_key()) {
+    // avoid conflicts with OS-context sent by a serverside SDK by using `contexts.client_os`
+    // instead of `contexts.os`. This is then preferred by the UI to show alongside device and
+    // browser context.
+    //
+    // Why not move the existing `contexts.os` into a different key on conflicts? Because we still
+    // want to index (derive tags from) the SDK-sent context.
+    let os_context_key = match event.platform.value().map(|x| x.as_str()) {
+        Some("javascript") => OsContext::default_key(),
+        _ => "client_os",
+    };
+
+    if is_known(os.family.as_str()) && !contexts.contains_key(os_context_key) {
         let version = get_version(&os.major, &os.minor, &os.patch);
-        contexts.add(Context::Os(Box::new(OsContext {
-            name: Annotated::from(os.family),
-            version: Annotated::from(version),
-            ..OsContext::default()
-        })));
+        contexts.insert(
```

Comment:
Please add a variant `Context::ClientOs(Box<OsContext>)` and return `"client_os"` from `fn Context::default_key`. That way, we only maintain this logic in one place and make it more reusable.

Reply:
Oh and I'm realizing that our enum derive only lowercases fields, where it should probably snake_case. Would that be possible to change with this PR?

Reply:
I don't think that is a good idea. the `Context` enum deals with context types (not default keys) but `client_os` is not its own context type yet.

I tried to implement this and it creates multiple possible representations for the same JSON: `{"contexts": {"os": ...}}` will be deserialized into an `OsContext` but can be serialized *from* a `ClientOs` variant as well.